Oauth signin

sophey opened the pull request:
Added a sign in window to pop up when the add button is pressed in the GUI. There is still no joint functionality, still working on getting the data after sign in and integrating that into the code.

sopheyd committed:
error with signin

sopheyd committed:
deleted unnecessary files, made runtime errors disappear, have signin button at localhost

still need to have it pop up when button is pressed

sopheyd committed:
add temporary open localhost when gui is run

sopheyd committed:
Merge remote-tracking branch 'google/master' into oauth-signin

sopheyd committed:
added pop up when you click "add", still no joint functionality

jkohen commented on bin/chat_server_log.log:
```diff
@@ -0,0 +1,4192 @@
+Apr 04, 2017 5:37:42 PM codeu.chat.ServerMain main
```
Probably a good idea not to submit log files to source control. You can add the file name to .gitignore to prevent this.

jkohen replied:
Ditto for .class and .lck files below. Source control doesn't excel at handling binary files, and it just makes the pull requests larger.

jkohen replied:
On the other hand, it's fine to store the button image, as it's small, and it makes sense to version images.

jkohen commented on client_secrets.json:
```diff
@@ -0,0 +1,9 @@
+{
+  "web": {
+    "client_id": "602021688774-lrtke5pp9kvekdpovvcamv5q5vo3hhfv.apps.googleusercontent.com",
+    "client_secret": "m6MddZ9oFfjS-cFbOVfnnNrc",
```
Nit, you probably wouldn't submit this in the clear to a public repository. I don't think it's an issue in the scope of this project.

jkohen commented on index.html:
```diff
@@ -0,0 +1,290 @@
+<!--
```
Nit, it looks like this came from https://github.com/googleplus/gplus-quickstart-python/blob/master/templates/index.html. It's good practice to mention that, so you can more easily find it in the future.

Have you made any changes to the original file?

jkohen commented on src/codeu/chat/client/simplegui/ChatSimpleGui.java:
```diff
@@ -22,6 +22,7 @@
 import codeu.chat.client.Controller;
 import codeu.chat.client.View;
 import codeu.chat.util.Logger;
+import codeu.chat.client.simplegui.Signin;
```
Since there are no other changes to the file, I suspect it's unused.

jkohen reviewed (commented):
Sorry for the three separate reviews, I'm new to this tool. That's all for now!

sopheyd committed:
Added gitignore file, got signin to vaguely work

User is able to sign in after pressing “add”, but no more than one user
is able to sign in.

sopheyd committed:
Merge branch 'review' into oauth-signin

sopheyd committed:
Merge branch 'review' into oauth-signin

sopheyd committed:
Changed Run_Server for no server

sopheyd committed:
Merge remote-tracking branch 'origin/oauth-signin' into oauth-signin

hallekfields committed:
added messageSecurity.java example for encryption

abbas22a committed:
Adding partially working image functionality

abbas22a committed:
Adding partially working image functionality
